includes/admin/class-wp-job-manager-promoted-jobs-admin.php

```diff
@@ -114,6 +128,54 @@ public function add_action_notice( $actions_handled ) {
 		return $actions_handled;
 	}
 
+	/**
+	 * Add the host of the Promote Job form to the array of allowed redirect hosts.
+	 *
+	 * @param array $hosts Allowed redirect hosts.
+	 * @return array Updated array of allowed redirect hosts.
+	 */
+	public function add_to_allowed_redirect_hosts( $hosts ) {
+		$hosts[] = wp_parse_url( WP_Job_Manager_Helper_API::get_api_base_url(), PHP_URL_HOST );
+		return $hosts;
+	}
+
+	/**
+	 * Handle the action to promote a job listing, validating as well as redirecting to the form on WPJobManager.com.
+	 *
+	 * @return void
+	 */
+	public function handle_promote_job() {
+		$post_id = absint( $_GET['post_id'] ?? 0 );
+		check_admin_referer( self::PROMOTE_JOB_ACTION . '-' . $post_id );
+		if ( ! $post_id ) {
+			wp_die( esc_html__( 'No job listing ID provided for promotion.', 'wp-job-manager' ), '', [ 'back_link' => true ] );
+		}
+		if ( ! is_ssl() ) {
+			wp_die( esc_html__( 'You must be using SSL to promote a job listing.', 'wp-job-manager' ), '', [ 'back_link' => true ] );
+		}
+		if ( ! current_user_can( 'manage_job_listings', $post_id ) || 'job_listing' !== get_post_type( $post_id ) ) {
+			wp_die( esc_html__( 'You do not have permission to promote this job listing.', 'wp-job-manager' ), '', [ 'back_link' => true ] );
+		}
+		$current_user = get_current_user_id();
+		$site_trust   = WP_Job_Manager_Site_Trust_Token::instance();
+		$token        = $site_trust->generate( 'user', $current_user );
+		if ( is_wp_error( $token ) ) {
+			wp_die( esc_html( $token->get_error_message() ) );
+		}
+		$url = add_query_arg(
+			[
+				'user_id'  => $current_user,
+				'job_id'   => $post_id,
+				'token'    => $token,
+				'site_url' => site_url( '', 'https' ),
+				'locale'   => get_user_locale( $current_user ),
+			],
+			WP_Job_Manager_Helper_API::get_api_base_url() . self::PROMOTE_JOB_FORM_PATH
```

Comment:
Just as a reminder, if we do it, it would also demand a change in the `add_to_allowed_redirect_hosts` method to get from the constant.

Reply:
The reason I didn't do this (used a constant) and instead preferred to make it dynamic based on `get_api_base_url()` is just to have...that "single point of change" for development.

I know that WP does provide us with the flexibility to change this easily, but then I already see three filters we'll need to change just to support a local WPJM environment:

1. wp_redirect (for this specific use case)
2. allowed_redirect_hosts (to allow the redirect above)
3. probably the HTTP request methods, to allow communicating with WPJM

In Sensei Pro, I know we have a single place/filter to change most this, so my idea was to bring some of that to here.

Maybe we could instead have a filter inside `get_api_base_url()` to make it easier to change instead of using the logic based on `-dev` versions?

Reply:
My main point was that we're getting the base API URL as the site URL. Currently, it's the same and works, but it doesn't seem semantic to me.

I just realized that we also have the private method `WP_Job_Manager_Com_API::get_api_base_url` (not related to this PR), which also seems a little weird. But just mentioning it because the path is to `wp-json` there, for example.

Anyway, if you're not comfortable to change that, I'm okay to keep it as it is.

Reply:
Ah, good catch! I see what you mean now. Thanks for the explanation!

I think I will unify the `get_api_base_url` in a way where it returns the `/wp-json` path for WPJobManager.com, while creating another `get_site_base_url()` where it returns only the base for WPJobManager.com.

I noticed that indeed we have two methods with the same name, and one of them has a filter and another doesn't. So...this will help improve the consistency too.

I'll share the commit changing this later. Thanks!

Reply:
Sorry, I forgot to document here, but the commits 4d16369, 28e2d23e2c958d8db1311db0328ff043b661df95 and 366cb88ff4ab04114e5d4353bf7be649d3d039c0 make this change.

Hopefully it is better nad it solves what you mentioned. ;)